Sources/SWBTaskExecution/TaskActions/ClangCompileTaskAction.swift

```diff
@@ -335,13 +338,15 @@ public final class ClangCompileTaskAction: TaskAction, BuildValueValidatingTaskA
                 } else {
                     files = nil
                 }
-                let diagnostics = moduleDependenciesContext.makeDiagnostics(files: files)
-                for diagnostic in diagnostics {
-                    outputDelegate.emit(diagnostic)
-                }
 
-                if diagnostics.contains(where: { $0.behavior == .error }) {
-                    return .failed
+                if let dependencyValidationOutputPath {
+                    let validationInfo = DependencyValidationInfo(files: files)
+                    _ = try executionDelegate.fs.writeIfChanged(
+                        dependencyValidationOutputPath,
+                        contents: ByteString(
+                            JSONEncoder(outputFormatting: .sortedKeys).encode(validationInfo)
+                        )
+                    )
```

Comment:
I'm a bit ambivalent on whether the validation action should just be directly consuming the trace files. It seems right for the Clang action to know about the implementation details here, but on the other hand it does seem a tiny bit wasteful to just process these files into other files?

Reply:
This seems like a good place to start! Easy to change later.